### PR TITLE
Update existing tests + Add tests for voting

### PR DESCRIPTION
**Pull request by Pieter12345**

### CODE REVIEW

File: test/SharedAssetTest.js [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+//        
+//        // Deposit money into the treasury.
+//        await asset.treasuryDeposit('InfoMessage', {from: user1, value: 1e12});
+//        
+//        // Payout to the shareholders.
+//        var currentTimeSec = Date.now() / 1000;
+//        await asset.payOwners({timestamp: currentTimeSec + parseInt(await asset.payoutPeriod)});
+//        
+//        // Assert that the owners have received the payout.
+//        // TODO - How to get Ether from address?
```
Comment (babeanu-dorian) [inline test/SharedAssetTest.js]: use [web3.eth.getBalance(address)](https://ethereum.stackexchange.com/questions/58482/getting-hold-of-web3-from-inside-a-truffle-js-test)
please validate balances after payments and share buying/selling

File: test/SharedAssetTest.js [hunk 2]
```diff
@@ -0,0 +1,562 @@
+/* global artifacts contract beforeEach it assert */
+
+const { assertRevert } = require('@aragon/test-helpers/assertThrow');
+const { hash } = require('eth-ens-namehash');
+const deployDAO = require('./helpers/deployDAO');
+
+const SharedAsset = artifacts.require('SharedAssetTestHelper.sol');
+const ANY_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+const functionIds = { // TODO - Get these from the contract TaskFunction enum instead.
```
Comment (babeanu-dorian) [inline test/SharedAssetTest.js]: either implement or remove this TODO

**Review by babeanu-dorian — CHANGES_REQUESTED** [review]

File: test/SharedAssetTest.js (hunk 2 above)
Comment (Pieter12345) [reply test/SharedAssetTest.js]: Removed TODO.

File: test/SharedAssetTest.js (hunk 1 above)
Comment (Pieter12345) [reply test/SharedAssetTest.js]: Re-implemented payout test and asserted that the seller is being paid when a buyer buys their offer. If a sell offer autocompletes, it's tough to assert the balance of the seller, as the gas price creates a quite large offset here.

**Review by babeanu-dorian — APPROVED** [review]

**Pull request merged**